Makefile.work

```diff
@@ -130,7 +141,12 @@ SONIC_BUILD_INSTRUCTION :=  make \
 .DEFAULT_GOAL :=  all
 
 %::
+ifneq (,$(filter $(CONFIGURED_ARCH), armhf arm64))
+	@$(DOCKER_MULTIARCH_CHECK)
+endif
 	@$(OVERLAY_MODULE_CHECK)
+	
+	@{ which j2 &> /dev/null && CONFIGURED_ARCH=$(CONFIGURED_ARCH) j2 $(SLAVE_DIR)/Dockerfile.j2 > $(SLAVE_DIR)/Dockerfile ; } || true
```

Comment:
>which j2 [](start = 4, length = 8)

Are you require j2 on the build server host? If yes, could you instead use the j2 inside slave container? I guess it is possible to pipe files from stdin

Reply:
This is required to build sonic slave container. If slave is NOT there how you can access the container files?

Reply:
Good point, you can ignore this ask.

---
In reply to: [296113832](https://github.com/Azure/sonic-buildimage/pull/2980#discussion_r296113832) [](ancestors = 296113832)